src/haywire/http_server.c

```diff
@@ -268,6 +288,80 @@ void http_stream_on_read_http_parser(uv_stream_t* tcp, ssize_t nread, const uv_b
+{
+    http_connection* connection = (http_connection*)tcp->data;
+
+    // TODO: nread == 0 should mean the socket closed. We should handle this properly.
+    if (nread >= 0)
+    {
+        //printf("BEFORE\tREAD: %d BUFF_LEN: %d PREV: %d\n", nread, connection->request_buffer_length, connection->prevbuflen);
+        
+        memcpy(&connection->request_buffer[connection->request_buffer_length], buf->base, nread);
+        connection->request_buffer_length += nread;
+        int counter = 0;
+
+        //printf("AFTER\tREAD: %d BUFF_LEN: %d PREV: %d ", nread, connection->request_buffer_length, connection->prevbuflen);
+        
+        while (connection->prevbuflen < connection->request_buffer_length)
+        {
+            counter++;
+            
+            int parsed;
+            char *method;
+            char *path;
+            int minor_version;
+            struct phr_header headers[100];
+            size_t method_len;
+            size_t path_len;
+            size_t num_headers;
+            num_headers = sizeof(headers) / sizeof(headers[0]);
+            
+            parsed = phr_parse_request(&connection->request_buffer[connection->prevbuflen], connection->request_buffer_length - connection->prevbuflen,
+                                       &method, &method_len, &path, &path_len, &minor_version, headers,
+                                       &num_headers, 0);
+
+            //printf("POS: %d PARSED: %d ----------------------------------------\n", connection->prevbuflen, parsed);
+            
+            if (parsed > 0)
+            {
+                // Successfully parsed the request.
+                connection->prevbuflen += parsed;
+                            
+                connection->keep_alive = 1;
+                http_request* request = create_http_request(connection);
+                connection->request = request;
+                
+                path[path_len] = 0x00;
+                request->url = path;
+                
+                http_request_complete_request(connection);
+            }
+            else if (parsed == -1)
+            {
+                memmove(connection->request_buffer, &connection->request_buffer[connection->prevbuflen], connection->request_buffer_length - connection->prevbuflen);
+                connection->request_buffer_length = connection->request_buffer_length - connection->prevbuflen;
+                connection->prevbuflen = 0;
+                
+                //printf("BUFF_LEN: %d PREV: %d\n", connection->request_buffer_length, connection->prevbuflen);
+                break;
+            }
+            else if (parsed == -2)
+            {
+                break;
+            }
+        }
+        
+        if (connection->prevbuflen == connection->request_buffer_length)
+        {
+            connection->request_buffer_length = 0;
+            connection->prevbuflen = 0;
+        }
+    }
```

Comment:
wrk or wrk2? Doesn't wrk2 run some requests to establish the overhead of the testing process it self?